.github/workflows/integration.yml

```diff
@@ -0,0 +1,290 @@
+name: Integration Gates
+on:
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  contents: read
+  packages: read
+  pull-requests: write
+
+jobs:
+  meta:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Assign author to PR
+        if: github.event.action == 'opened'
+        uses: toshimaru/auto-author-assign@v2.1.1
+
+      - name: Validate PR title
+        uses: amannn/action-semantic-pull-request@v5
+        with:
+          requireScope: false
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Assign labels to PR
+        uses: mauroalderete/action-assign-labels@v1
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          conventional-commits: |
+            conventional-commits:
+              - type: 'fix'
+                nouns: ['FIX', 'Fix', 'fix', 'FIXED', 'Fixed', 'fixed']
+                labels: ['fix']
+              - type: 'feature'
+                nouns: ['FEATURE', 'Feature', 'feature', 'FEAT', 'Feat', 'feat']
+                labels: ['feature']
+          maintain-labels-not-matched: false
+          apply-changes: true
+  fitness-compliance:
+    runs-on: ubuntu-latest
+    needs: meta
+    steps:
+      - uses: actions/checkout@v4
+      - name: Dependency review
+        uses: actions/dependency-review-action@v4
+        with:
+          license-check: true
+          deny-licenses: 'AGPL-1.0-or-later, GPL-1.0-or-later, LGPL-2.0-or-later, NGPL'
+          vulnerability-check: false
+          comment-summary-in-pr: on-failure
+
+      - name: Changelog review
+        if: contains(github.event.pull_request.labels.*.name, 'feature') || contains(github.event.pull_request.labels.*.name, 'fix')
+        run: |
+          gh pr diff "$BRANCH" --name-only >> diff.txt
+          grep -Fxq "docs/changelog/changelog_en.md" diff.txt
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          BRANCH: ${{ github.head_ref }}
+  fitness-code-quality:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Node and Yarn
+        uses: ./.github/actions/setup
+        with:
+          npm-auth-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Run prettier
+        run: yarn format
+
+      - name: Run linter
+        run: yarn lint
+  test-unit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Node and Yarn
+        uses: ./.github/actions/setup
+        with:
+          npm-auth-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Unit tests
+        run: |
+          pushd ${{ github.workspace }}/apps/server/
+          yarn start &
```

Comment:
why do we need yarn start here? Is it because of the `IntersectionObserver is not defined` ? we can keep it for now, but I would like to investigate, if we really need that.

Reply:
I don't remember why we needed to do this at the time, but the unit tests pass without starting the server now.